### PR TITLE
Tweak: Hidden Elements Behavior - Add the ability to select hidden elements behavior [ED-2963]

### PR DESCRIPTION
**Pull request by guyc-E**
…ments behvior

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/js/editor/components/settings/editor-preferences/manager.js [hunk 1]
```diff
@@ -14,7 +14,10 @@ export default class extends BaseManager {
 			ui_theme: this.onUIThemeChanged,
 			panel_width: this.onPanelWidthChanged,
 			edit_buttons: this.onEditButtonsChanged,
+			show_hidden_elements: this.OnShowHiddenElementsChange,
 		};
+
+		elementor.on( 'globals:loaded', this.OnShowHiddenElementsChange );
```
Comment (udidol) [inline assets/dev/js/editor/components/settings/editor-preferences/manager.js:20]: Don't run this here, add the appropriate classes where body classes for the Preview section are added.

**Review by udidol — CHANGES_REQUESTED** [review]

File: includes/frontend.php [hunk 2]
```diff
@@ -298,6 +298,14 @@ public function body_class( $classes = [] ) {
...
 		}
 
+		$editor_preferences = SettingsManager::get_settings_managers( 'editorPreferences' );
+		
+		$show_hidden_elements = $editor_preferences->get_model()->get_settings( 'show_hidden_elements' );
+
+		if( 'yes' === $show_hidden_elements ){
+			$classes[] = 'show-hidden-elements';
+		}
+
```
Comment (udidol) [inline includes/frontend.php:308]: This should only happen in the Editor, not in the frontend.
Try adding a check for `Plugin::$instance->editor->is_edit_mode()`

**Review by udidol — CHANGES_REQUESTED** [review]

File: includes/frontend.php [hunk 3]
```diff
@@ -298,6 +298,18 @@ public function body_class( $classes = [] ) {
 			$classes[] = 'elementor-page elementor-page-' . $id;
 		}
 
+		$is_preview_mode = Plugin::$instance->preview->is_preview_mode();
+
+		if( $is_preview_mode ){
```
Comment (udidol) [inline includes/frontend.php:303]:
```suggestion
		if( Plugin::$instance->preview->is_preview_mode() ){
```

**Review by udidol — APPROVED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/settings/editor-preferences/manager.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ export default class extends BaseManager {
 			ui_theme: this.onUIThemeChanged,
 			panel_width: this.onPanelWidthChanged,
 			edit_buttons: this.onEditButtonsChanged,
+			show_hidden_elements: this.OnShowHiddenElementsChange,
```
[the remark this answers is not in the document]
Comment (guyc-E) [reply assets/dev/js/editor/components/settings/editor-preferences/manager.js:17]: I think you forgot to change your suggested code. it looks exactly the same

File: assets/dev/js/editor/components/settings/editor-preferences/manager.js [hunk 5]
```diff
@@ -63,4 +64,8 @@ export default class extends BaseManager {
 		// Let the button change before the high-performance action of rendering the entire page
 		setTimeout( () => elementor.getPreviewView()._renderChildren(), 300 );
 	}
+
+	OnShowHiddenElementsChange() {
+		elementor.$previewContents.find( 'body' ).toggleClass( 'e-preview--show-hidden-elements' );
```
Comment (kobizz) [inline assets/dev/js/editor/components/settings/editor-preferences/manager.js:69]: Use `elementorFrontend.elements.$body` instead

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**